python/cudf_polars/cudf_polars/dsl/expressions/aggregation.py

```diff
@@ -156,22 +164,44 @@ def agg_request(self) -> plc.aggregation.Aggregation:  # noqa: D102
     def _reduce(
         self, column: Column, *, request: plc.aggregation.Aggregation, stream: Stream
     ) -> Column:
-        if (
-            self.name in {"mean", "median"}
-            and plc.traits.is_fixed_point(column.dtype.plc_type)
-            and self.dtype.plc_type.id() in {plc.TypeId.FLOAT32, plc.TypeId.FLOAT64}
+        is_mean_or_median = self.name in {"mean", "median"}
+        is_quantile = self.name == "quantile"
+
+        out_dtype = self.dtype
```

Comment:
So Polars has more nuanced casting rules for different aggregations now than it used to?